### ISSUE TITLE
Frontend — TypeError: "o is undefined"

### ISSUE THREAD

**Issue by selfagency** [strapi/strapi#3098]
**Informations**
- **Node.js version**: 10.15.3
- **NPM version**: 6.9.0
- **Strapi version**: 3.0.0-alpha.25.2
- **Database**: MySQL
- **Operating system**: Ubuntu 18.04
- **(Optional) Link to your Project**: https://gitlab.com/selfagencyllc/kifa-cms

**What is the current behavior?**

In development, it works fine. In production, when attempting to add a new user, I get a blank screen and the following console error:

```
TypeError: "o is undefined"
    renderAttr main.js:48
    value main.js:48
    Co main.js:16
    Oo main.js:16
    Bo main.js:16
    wn main.js:16
    An main.js:16
    Wn main.js:16
    Hn main.js:16
    Bn main.js:16
    En main.js:16
    enqueueSetState main.js:24
    setState main.js:40
    onStateChange main.js:72
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    l main.js:45
    n main.js:37
    e main.js:45
    dispatch main.js:45
    F main.js:45
    x main.js:45
    n main.js:45
    s main.js:45
    r main.js:45
    x main.js:45
    b main.js:45
    h main.js:45
    o main.js:45
main.js:16
    tn main.js:16
    callback main.js:16
    Zo main.js:16
    Xo main.js:16
    yn main.js:16
    bn main.js:16
    qn main.js:16
    unstable_runWithPriority main.js:53
    qn main.js:16
    Wn main.js:16
    Hn main.js:16
    Bn main.js:16
    En main.js:16
    enqueueSetState main.js:24
    setState main.js:40
    onStateChange main.js:72
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    notify main.js:37
    notifyNestedSubs main.js:37
    onStateChange main.js:37
    l main.js:45
    n main.js:37
    e main.js:45
    dispatch main.js:45
    F main.js:45
    x main.js:45
    n main.js:45
    s main.js:45
    r main.js:45
    x main.js:45
    b main.js:45
    h main.js:45
    o main.js:45
```

**Steps to reproduce the problem**
Navigate to `https://YOURSITE/admin/plugins/content-manager/user/create?redirectUrl=/plugins/content-manager/user?_limit=10&_page=1&_sort=id&source=users-permissions`

**What is the expected behavior?**
For the page to load.

**Comment by derrickmehaffy** [comment]
@selfagency Are you using a proxy? If so there is a bug with a header rewrite that causes this. I know of a fix for Nginx but not for apache off the top of my head.

**Comment by selfagency** [comment]
I am using Nginx.

**Comment by soupette** [comment]
Duplicate of https://github.com/strapi/strapi/issues/2424

**Comment by derrickmehaffy** [comment]
@selfagency this fix is what you need: https://github.com/strapi/strapi/issues/2424#issuecomment-467851988

@soupette I'll keep this one open since the other duplicate got closed.

**Comment by selfagency** [comment]
I implemented the Nginx config above with no improvement. Still getting the same error.

**Comment by selfagency** [comment]
Never mind I missed a change. It works now. Thanks.

**Comment by milanffm** [comment]
I have the same problem with a fresh installation:
3.0.0-alpha.25.2.
DB: Mysql

I contact my hoster, and they can't fix it for me. 
I temporary use the fix from @broTrue 

 \plugins\users-permissions\models\User.settings.json

"password": {
      "type": "password",
      "minLength": 6,
      "configurable": false,
      "private": true
    },`


`privat: false`

is this really unseure?

**Comment by derrickmehaffy** [comment]
@milanffm *ABSOLUTELY* Do not do this, you are effectively allowing for your password hashes to be grabbed from an API route that has a relation with users and/or if at any point someone can query your users, you just provided them with a way to grab and crack these hashes offline.

PLEASE do not change this, I would really hate to see someone from the Strapi community be involved in leaking users password even without meaning to.

**Comment by milanffm** [comment]
but what can I  do, when I don't have access to the apache config? 
To you know a shared host, where I can use strapi? I don't whant to setup my own server.

**Comment by derrickmehaffy** [comment]
@milanffm to be frank, that is probably going to be your best option. Strapi is not designed to be ran on a shared host like that (especially one just built for static web files). It's a service to be ran on a server.

**Comment by SamanthaTian** [comment]
I have the same situation as milanffm.
3.0.0-alpha.25.2.
DB: Mysql

shared host, no access to the apache config but can use .htaccess.
Any walk around?

**Comment by milanffm** [comment]
I think their is no solution for this, until they fix the bug. I set up my one server, to have access to the nginx config. 
But you can only move the backend to the server and leave all frontend/Email stuff by the shared hoster. Than you don't have to set up a mailserver and all this email shit :-)

**Comment by soupette** [comment]
Fixed in #3633...
